bpf/process/bpf_gethostpid.c

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: GPL-2.0
+/* Copyright Authors of Tetragon */
+
+#include "vmlinux.h"
+
+#include "bpf_helpers.h"
+#include "bpf_tracing.h"
+
+char _license[] __attribute__((section("license"), used)) = "GPL";
+
+struct hostpid_filter {
+	__s32 fd;
+	__s32 whence;
+	__s64 off;
+};
+
+struct hostpid_event {
+	__s32 pid;
+};
+
+struct {
+	__uint(type, BPF_MAP_TYPE_ARRAY);
+	__type(key, __u32);
+	__type(value, struct hostpid_filter);
+	__uint(max_entries, 1);
+} tg_hpid_filter_map SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
+	__type(key, int);
+	__type(value, struct hostpid_event);
+	__uint(max_entries, 128);
+} tg_hpid_event_map SEC(".maps");
+
+/* helper program to retrieve tetragon's host pid via an lseek bogus call */
+__attribute__((section(("kprobe/lseek_hook")), used)) int
+tg_hpid_hook(struct pt_regs *ctx)
+{
+	int zero = 0;
+	struct hostpid_event ev;
+	struct hostpid_filter *f;
+	struct pt_regs *ctx_;
+
+	f = map_lookup_elem(&tg_hpid_filter_map, &zero);
+	if (!f)
+		return 0;
+
+	ctx_ = PT_REGS_SYSCALL_REGS(ctx);
+	if (PT_REGS_PARM1_CORE_SYSCALL(ctx_) != f->fd)
+		return 0;
+
+	if (PT_REGS_PARM2_CORE_SYSCALL(ctx_) != f->off)
+		return 0;
+
+	if (PT_REGS_PARM3_CORE_SYSCALL(ctx_) != f->whence)
+		return 0;
+
+	ev.pid = get_current_pid_tgid() >> 32;
+	perf_event_output(ctx, &tg_hpid_event_map, BPF_F_CURRENT_CPU, &ev, sizeof(ev));
+	return 0;
```

Comment:
seems like too much (using perf rb) for retrieving just one pid value.. could we just update map? or perhaps something like in `pkg/sensors/map_update.go ` maybe the ebpf program could return the pid as return value, but I might be too optimistic ;-)

Reply:
I used a perf rb to handle the case where we get >1 than one events. it's very unlikely to happen but it does not seem like a lot of trouble to handle it.

Reply:
> maybe the ebpf program could return the pid as return value, but I might be too optimistic ;-)

That's what I tried first using BPF_PROG_RUN, but I couldn't make it to work. Specifically, I could not find a program type that would allow me to get the pid. So, instead, I used a system call. Is there  a way to use `get_current_pid_tgid()` from a bpf program that we BPF_PROG_RUN?